#2307 Implement NonElectrical blocks WT IEC

GerardThevenon opened the pull request:
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

cossartque commented:
In branch 1533, the following models have been added in NonElectrical : FirstOrderRamp, FirstOrderRampLimit, IntegratorFixLimits, IntegratorVariableLimits, DelayFive, MultiSwitchFive, MultiSwitchFour, MultiSwitchThree, SwitchInt.

It seems that some of them are in fact not needed as you only implemented the following models : FallingEdgeDetection, FirstOrderFilter, FirstOrderFilterLimitationDetection, RisingEdgeDetection.

In the document IEC, the following blocks are presented in the appendix D : Switch (with N possible inputs), Time step delay, Stand-alone ramp rate limiter, First order filter, Lookup table, Comparator, Timer, Anti windup integrator, Integrator with reset, First order filter with limitation detection, Rising edge detection, Falling edge detection, Delay flag, Variable delay flag, Dead band,  Circuit breaker.

Could you please tell why some models in 1533 are in fact not needed, what are the names correspondance, why some models in the document IEC are not needed and what are the names in the MSL in that case.

Thanks.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/CMakeLists.txt:
```diff
@@ -11,6 +11,9 @@
 set(MODEL_FILES
   package.mo
   package.order
+  FallingEdgeDetection.mo
```
I am not always sure if each block should be in Continuous or in NonLinear

GerardThevenon replied:
Yes, let's discuss that.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FallingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block FallingEdgeDetection "Output y is true, if the input u has fallen to a lower value since the previous sample"
```
What is the difference with the FallingEdge blocks in the MSL Logical and/or MathBoolean ?

cossartque replied:
Is it the tS ?

GerardThevenon replied:
As you saw, the MSL blocks take a boolean input.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FallingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+
+  parameter Types.Time tS "Integration time step in s";
+
+  Modelica.Blocks.Interfaces.RealInput u "Input signal connector" annotation(
```
No start value for u and y ?

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FallingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+    Line(points = {{-120, 0}, {-10, 0}}, color = {0, 0, 127}));
+  connect(fixedDelay.y, less.u2) annotation(
+    Line(points = {{-38, -40}, {-20, -40}, {-20, -8}, {-10, -8}}, color = {0, 0, 127}));
+
```
Please add a preferred view

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilter.mo:
```diff
@@ -0,0 +1,75 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block FirstOrderFilter "First order filter with absolute and rate limits, and a freezing flag"
```
What is the difference with RateLimFirstOrderFreeze ?
Also, could the name reflect more what is included ? It is not a simple FirstOrderFilter, it includes limits, rate lims, freeze

GerardThevenon replied:
RateLimFirstOrderFreeze seems to be indeed a good model for what is required by the IEC norm. The interfaces are optional, the behaviour is implemented in the text view, I think the 1/T factor is not at the same place.

Let's change the name in a further push.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilter.mo:
```diff
@@ -0,0 +1,75 @@
+
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Types.PerUnit DuMax "Maximun ramp rate";
```
Shouldn't it be DyMax and DyMin ?

GerardThevenon replied:
Yes.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilter.mo:
```diff
@@ -0,0 +1,75 @@
+
+  parameter Types.PerUnit DuMax "Maximun ramp rate";
+  parameter Types.PerUnit DuMin(start = -DuMax) "Minimun ramp rate";
+  parameter Types.Time tI(start = 1) "Filter time constant in s";
```
What about the case where tI = 0 ? Should we have another model for that too ?

GerardThevenon replied:
I made another model.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilter.mo:
```diff
@@ -0,0 +1,75 @@
+  parameter Types.PerUnit YMax "Upper limit of output";
+  parameter Types.PerUnit YMin = -YMax "Lower limit of output";
+
+  Modelica.Blocks.Interfaces.RealInput u annotation(
```
No start value for u and y ?

GerardThevenon replied:
I don't think u_start is needed ?

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilterLimitationDetection.mo:
```diff
@@ -0,0 +1,68 @@
+  import Modelica;
+  import Dynawo;
+
+  extends Dynawo.NonElectrical.Blocks.Continuous.FirstOrderFilter;
```
Nice!

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilterLimitationDetection.mo:
```diff
@@ -0,0 +1,68 @@
+    Line(points = {{152, 40}, {154, 40}, {154, 72}, {158, 72}}, color = {255, 0, 255}));
+  connect(greater.y, or1.u1) annotation(
+    Line(points = {{42, 80}, {158, 80}}, color = {255, 0, 255}));
+
```
No preferred view ? Is it done by the extends ?

GerardThevenon replied:
It is not done by the extends, so I added it.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FirstOrderFilterLimitationDetection.mo:
```diff
@@ -0,0 +1,68 @@
+    Placement(visible = true, transformation(origin = {210, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanOutput fMin annotation(
+    Placement(visible = true, transformation(origin = {210, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Logical.Greater greater annotation(
```
Are you sure the greater are well connected ? Do we have fMax = true when we reach the upper limits ? Same questions for lessEqual

GerardThevenon replied:
I made a mistake, it is now corrected.

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/RisingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+
+block RisingEdgeDetection "Output y is true, if the input u has risen to a higher value since the previous sample"
+  import Modelica;
+
```
import Dynawo missing

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FallingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+
+block FallingEdgeDetection "Output y is true, if the input u has fallen to a lower value since the previous sample"
+  import Modelica;
+
```
import Dynawo missing

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/FallingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Types.Time tS "Integration time step in s";
```
Dynawo.Types.Time

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/RisingEdgeDetection.mo:
```diff
@@ -0,0 +1,42 @@
+
+  parameter Dynawo.Types.Time tS "Integration time step in s";
+
+  Modelica.Blocks.Interfaces.RealInput u "Input signal connector" annotation(
```
No start values for u and y ?

GerardThevenon commented:
> In the document IEC, the following blocks are presented in the appendix D : 

1. Switch (with N possible inputs) : done in 1533, MultiSwitchFive, MultiSwitchFour, MultiSwitchThree, but in the next push, I am adding a new, improved version of MultiSwitch
2. Time step delay : not needed (FixedDelay in MSL)
3. Stand-alone ramp rate limiter : not needed for now (SlewRateLimiter in MSL)
4. First order filter : done in 1533, FirstOrderRamp, FirstOrderRampLimit, but a little different from the description, so I have added FirstOrderFilter
5. Lookup table : not needed (CombiTable1Ds in MSL)
6. Comparator : not needed (Greater, GreaterEqual, Less, LessEqual in MSL)
7. Timer : not needed (Timer in MSL), but opposite logic
8. Anti windup integrator : done in 1533, IntegratorFixLimits, IntegratorVariableLimits, but a little different from the description, so in the next push, I am adding AntiWindupIntegrator
9. Integrator with reset : needed because it is slightly different to LimIntegrator (MSL), in the next push, I am adding IntegratorWithReset
10. First order filter with limitation detection : needed, I have added FirstOrderWithLimitation
11. Rising edge detection : needed, I have added RisingEdgeDetection
12. Falling edge detection : needed, I have added FallingEdgeDetection
13. Delay flag : done in 1533, DelayFlag, thanks to SwitchInt, these models are OK, I am just making some minor changes in the next push
14. Variable delay flag : needed, in the next push, I am adding VariableDelayFlag
15. Dead band : not needed (DeadZone in MSL)
16. Circuit breaker : not needed (IdealOpeningSwitch in MSL)

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/CMakeLists.txt:
```diff
@@ -11,6 +11,12 @@
   package.order
+  AntiWindupIntegrator.mo
+  FallingEdgeDetection.mo
+  FirstOrderFilter.mo
```
AbsLimRateLimFirstOrderFreeze

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/CMakeLists.txt:
```diff
@@ -11,6 +11,12 @@
+  AntiWindupIntegrator.mo
+  FallingEdgeDetection.mo
+  FirstOrderFilter.mo
+  FirstOrderFilterByPass.mo
```
AbsLimRateLimFeedthroughFreeze

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFeedthroughFreeze.mo:
```diff
@@ -0,0 +1,63 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block AbsLimRateLimFeedthroughFreeze "First order filter with absolute and rate limits, and a freezing flag"
```
Please explain here in the comment the difference with AbsLimRateLimFirstOrderFreeze

cossartque replied:
It is in fact not a first order but a feedthrough

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFeedthroughFreeze.mo:
```diff
@@ -0,0 +1,63 @@
+  parameter Types.PerUnit DyMax "Maximun ramp rate";
+  parameter Types.PerUnit DyMin(start = -DyMax) "Minimun ramp rate";
+  parameter Types.Time tS "Integration time step in s";
+  parameter Types.PerUnit Y0 "Initial value of output";
```
This one should be just before equation

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFirstOrderFreeze.mo:
```diff
@@ -0,0 +1,75 @@
+  parameter Types.PerUnit DyMax "Maximun ramp rate";
+  parameter Types.PerUnit DyMin(start = -DyMax) "Minimun ramp rate";
+  parameter Types.Time tI(start = 1) "Filter time constant in s";
+  parameter Types.PerUnit Y0 "Initial value of output";
```
This one should be just before equation

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFeedthroughFreeze.mo:
```diff
@@ -0,0 +1,63 @@
+
+block AbsLimRateLimFeedthroughFreeze "First order filter with absolute and rate limits, and a freezing flag"
+  import Modelica;
+
```
import Dynawo.Types is missing

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFirstOrderFreeze.mo:
```diff
@@ -0,0 +1,75 @@
+
+block AbsLimRateLimFirstOrderFreeze "First order filter with absolute and rate limits, and a freezing flag"
+  import Modelica;
+
```
import Dynawo.Types is missing

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFirstOrderFreezeLimDetection.mo:
```diff
@@ -0,0 +1,69 @@
+
+  extends Dynawo.NonElectrical.Blocks.Continuous.AbsLimRateLimFirstOrderFreeze;
+
+  Modelica.Blocks.Interfaces.BooleanOutput fMax annotation(
```
Please add a comment for fMax and fMin

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFirstOrderFreezeLimDetection.mo:
```diff
@@ -0,0 +1,69 @@
+
+  Modelica.Blocks.Interfaces.BooleanOutput fMax annotation(
+    Placement(visible = true, transformation(origin = {210, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {110, 60}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanOutput fMin annotation(
```
Blank line after fMin

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFirstOrderFreezeLimDetection.mo:
```diff
@@ -0,0 +1,69 @@
+    Placement(visible = true, transformation(origin = {30, 80}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Logical.Greater greater1 annotation(
+    Placement(visible = true, transformation(origin = {30, -80}, extent = {{-10, 10}, {10, -10}}, rotation = 0)));
+  Modelica.Blocks.Logical.LessEqual lessEqual annotation(
```
Are you sure it is lessEqual here and the line after ? Thanks

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AbsLimRateLimFirstOrderFreezeLimDetection.mo:
```diff
@@ -0,0 +1,69 @@
+    Line(points = {{122, 0}, {124, 0}, {124, -32}, {128, -32}}, color = {0, 0, 127}, pattern = LinePattern.Dash));
+  connect(integrator.y, lessEqual1.u1) annotation(
+    Line(points = {{82, 0}, {90, 0}, {90, -40}, {128, -40}}, color = {0, 0, 127}, pattern = LinePattern.Dash));
+  annotation(
```
Blank line before annotation

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AntiWindupIntegrator.mo:
```diff
@@ -0,0 +1,91 @@
+
+block AntiWindupIntegrator "Integrator with anti windup and anti winddown"
+  import Modelica;
+
```
import Dynawo.Types missing

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AntiWindupIntegrator.mo:
```diff
@@ -0,0 +1,91 @@
+  parameter Types.PerUnit DyMax "Maximun ramp rate";
+  parameter Types.PerUnit DyMin(start = -DyMax) "Minimun ramp rate";
+  parameter Types.Time tI "Filter time constant in s";
+  parameter Types.PerUnit Y0 "Initial value of output";
```
This one should be just before equation

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AntiWindupIntegrator.mo:
```diff
@@ -0,0 +1,91 @@
+
+  Modelica.Blocks.Interfaces.RealInput u annotation(
+    Placement(visible = true, transformation(origin = {-220, 1.77636e-15}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -1.77636e-15}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.BooleanInput fMax(start = false) annotation(
```
Please add a comment for fMax and fMin

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/AntiWindupIntegrator.mo:
```diff
@@ -0,0 +1,91 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block AntiWindupIntegrator "Integrator with anti windup and anti winddown"
```
with absolute and rate limits, anti windup and anti winddown

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/DecreaseDetection.mo:
```diff
@@ -0,0 +1,47 @@
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Dynawo.Types.Time tS "Integration time step in s";
+  parameter Real U0 "Inital input signal value";
```
Should be just before equation. Also "Initial"

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/IntegratorWithReset.mo:
```diff
@@ -0,0 +1,60 @@
+
+block IntegratorWithReset "First order filter with absolute and rate limits, and a freezing flag"
+  import Modelica;
+
```
import Dynawo.Types missing

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/IntegratorWithReset.mo:
```diff
@@ -0,0 +1,60 @@
+
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Types.Time tI(start = 1) "Filter time constant in s";
```
Why the start = 1 ?

cossartque replied:
Also it is not a filter but an integrator

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/IntegratorWithReset.mo:
```diff
@@ -0,0 +1,60 @@
+  extends Modelica.Blocks.Icons.Block;
+
+  parameter Types.Time tI(start = 1) "Filter time constant in s";
+  parameter Types.PerUnit Y0 "Initial value of output";
```
This should be just before equation

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/IntegratorWithReset.mo:
```diff
@@ -0,0 +1,60 @@
+  Modelica.Blocks.Logical.Switch switch1 annotation(
+    Placement(visible = true, transformation(origin = {70, 0}, extent = {{-10, 10}, {10, -10}}, rotation = 0)));
+
+
```
Only one line here

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/Continuous/IntegratorWithReset.mo:
```diff
@@ -0,0 +1,60 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block IntegratorWithReset "First order filter with absolute and rate limits, and a freezing flag"
```
Integrator with absolute limits and reset

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/DelayFlag.mo:
```diff
@@ -0,0 +1,86 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block DelayFlag
```
Is it possible to have a small description ?

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/DelayFlag.mo:
```diff
@@ -0,0 +1,86 @@
+  import Dynawo;
+  import Dynawo.Types;
+
+  parameter Types.PerUnit tD "Delay flag time constant, specifies the duration F0 will keep the value 2";
```
I think this comment could be clearer, thanks

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/MultiSwitch.mo:
```diff
@@ -0,0 +1,36 @@
+
+block MultiSwitch
+  import Modelica;
+  import Dynawo;
```
useless

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/MultiSwitch.mo:
```diff
@@ -0,0 +1,36 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block MultiSwitch
```
Please add a comment, thanks

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/MultiSwitch.mo:
```diff
@@ -0,0 +1,36 @@
+  Modelica.Blocks.Interfaces.RealOutput y annotation(
+    Placement(transformation(extent={{100,-17},{134,17}})));
+
+
```
One line

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/SwitchInteger.mo:
```diff
@@ -0,0 +1,43 @@
+
+  Modelica.Blocks.Interfaces.IntegerInput u0 "Connector of second Real input signal" annotation(
+    Placement(visible = true, transformation(origin = {-120, -40}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.IntegerInput u1 "Connector of first Real input signal" annotation(
```
the position of u1 in the diagram view is not symmetrical with u0

cossartque replied:
Not Real but Integer

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/SwitchInteger.mo:
```diff
@@ -0,0 +1,43 @@
+*/
+
+block SwitchInteger "Switch between two Integer signals"
+
```
no line

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/SwitchInteger.mo:
```diff
@@ -0,0 +1,43 @@
+
+  import Modelica;
+
+  Modelica.Blocks.Interfaces.IntegerInput u0 "Connector of second Real input signal" annotation(
```
Not Real but Integer

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/VariableDelayFlag.mo:
```diff
@@ -0,0 +1,107 @@
+
+  Modelica.Blocks.Interfaces.BooleanInput fI annotation(
+    Placement(visible = true, transformation(origin = {-120, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -1.77636e-15}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput tD "Delay flag time constant, specifies the duration F0 will keep the value 2" annotation(
```
Comment could be clearer

cossartque commented on dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/VariableDelayFlag.mo:
```diff
@@ -0,0 +1,107 @@
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block VariableDelayFlag
```
Small comment ?
